tests/integration/jpa/appl/src/main/java/io/helidon/tests/integration/jpa/appl/Dispatcher.java

```diff
@@ -106,44 +110,48 @@ private Handle getHandle(final String name) {
         }
         final String className = name.substring(0, serpPos);
         final String methodName = name.substring(serpPos + 1, nameLen);
-        if ("InsertIT".equals(className)) {
-            try {
-                return new Handle(
-                        insertIt,
-                        InsertIT.class.getDeclaredMethod(methodName, TestResult.class),
-                        result);
-            } catch (NoSuchMethodException ex) {
-                result.throwed(ex);
-            }
-        } else if ("UpdateIT".equals(className)) {
-            try {
-                return new Handle(
-                        updateIt,
-                        UpdateIT.class.getDeclaredMethod(methodName, TestResult.class),
-                        result);
-            } catch (NoSuchMethodException ex) {
-                result.throwed(ex);
-            }
-        } else if ("DeleteIT".equals(className)) {
-            try {
-                return new Handle(
-                        deleteIt,
-                        DeleteIT.class.getDeclaredMethod(methodName, TestResult.class),
-                        result);
-            } catch (NoSuchMethodException ex) {
-                result.throwed(ex);
-            }
-        } else if ("QueryIT".equals(className)) {
-            try {
-                return new Handle(
-                        queryIt,
-                        QueryIT.class.getDeclaredMethod(methodName, TestResult.class),
-                        result);
-            } catch (NoSuchMethodException ex) {
-                result.throwed(ex);
-            }
-        } else {
+        if (null == className) {
             result.fail("Unknown test class: " + className);
+        } else switch (className) {
+            case "InsertIT":
```

Comment:
This seems too complicated.
Maybe extract to method and catch the `NoSuchMethodException` just once?

Reply:
This will be completely redesigned when I'll be writing native image tests for DB Client. In this case I just wanted to find some working solution to simply run tests in MP application environment and return results.
Next step is to make it jUnit like as much as possible and scan all classes for specific annotated classes/methods so this whole piece of code will be removed.
I'll extract method for now to make it better readable.